### PR TITLE
Separate src directory

### PR DESCRIPTION
**Pull request by amarrella**
Since updating hashes for local files is not a great workflow,  I created a `src`directory where the changes need to actually happen without hashes for local files.

The main directory still contains the hashed files (better for performance).

`make compile` takes the files from src into the main directory, and re-freezes them. Compilation takes about ~3 minutes but it's necessary only at the end of the workflow to generate properly hashed files.

### CODE REVIEW

File: kubernetes/argo/patch.diff [hunk 1]
```diff
@@ -3285,8 +3285,7 @@ index 0aee00a..c0849f2 100644
 +++ b/kubernetes/package.dhall
 @@ -10,4 +10,7 @@
  , argocd =
-       ./argocd/package.dhall sha256:b0ee43f5521aa371be2ea6ac13e1a1c85c7f01cec15431ce58ec79fe9f740847
-     ? ./argocd/package.dhall
+       ./argocd/package.dhall
```
Comment (retnuh) [inline kubernetes/argo/patch.diff]: did you mean to edit the patch?

File: scripts/freeze.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-for file in $(find . -name '*.dhall' -print | perl -n -e '$x = $_; $x =~ tr%/%%cd; print length($x), " $_";' | sort -rn -k 1 | awk '{print $2}'); do
+for file in $(find . -path ./src -prune -o -name '*.dhall' -print | perl -n -e '$x = $_; $x =~ tr%/%%cd; print length($x), " $_";' | sort -rn -k 1 | awk '{print $2}'); do
```
Comment (retnuh) [inline scripts/freeze.sh]: we may be able to replace this with a `dhall resolve --transitive-dependencies` in the src repo now that the cache doesn't interfere.  I'll co the branch and have a go.

File: src/kubernetes/argo/patch.diff [hunk 3]
```diff
@@ -0,0 +1,3302 @@
+diff --git a/kubernetes/argo/defaults.dhall b/kubernetes/argo/defaults.dhall
```
Comment (retnuh) [inline src/kubernetes/argo/patch.diff]: is there any point in having this patch file here?

Comment (amarrella) [reply src/kubernetes/argo/patch.diff]: It was to keep the changes we did for argo workflow visible and make it usable with `git patch`

File: kubernetes/argo/patch.diff (hunk 1 above)
Comment (amarrella) [reply kubernetes/argo/patch.diff]: no, will fix, thanks

File: src/kubernetes/argo/patch.diff (hunk 3 above)
Comment (retnuh) [reply src/kubernetes/argo/patch.diff]: Right but does it make sense to have it in both places?

File: scripts/freeze.sh (hunk 2 above)
Comment (retnuh) [reply scripts/freeze.sh]: dangit at first blush looks like the resolver doesn't keep the order, i.e. the files listed at top of output have dependencies in middle somewhere :exploding_head:

File: src/kubernetes/argo/patch.diff (hunk 3 above)
Comment (amarrella) [reply src/kubernetes/argo/patch.diff]: For sure it makes sense in `src/`. I'll look into removing it from `/kubernetes`

**Comment by amarrella** [comment]
>  Just wondering if it would be better to generate to a directory, and have the toplevel package.dhall file just take on the generated one, to remove the level of indirection?

At this point we have a lot of references to packages directly (using the root `package.dhall` is generally a bad idea as it pull down all the packages, it's only useful for CI pretty much)

**Pull request merged**